crates/kilt-dip-primitives/src/merkle/v0/provider_state/tests.rs

```diff
@@ -0,0 +1,566 @@
+		let parachain_state_root: H256 =
+			hex!("3a27f8d59c8bcc51bf3735ecdc0ce1304127a5b9e707e956e22633179493d55c").into();
+		let dip_commitment_proof = DipCommitmentStateProof(vec![
+			hex!("7f240d1c9ddb0ba4c0507243fb24936031b342ed12d34ad897938af6739bf519bdc2f101d67bf7dac85501a12dfa1fa4ab9a0000804aba9a5555257d6477fc5a74aaad1eaa24543e7eb1b4ac5ff1c00a50f6e63b3e").to_vec(),
+			hex!("800c808052d9b1ca86bf39ca4b7d574a5bcea35625200b5ff30c4517f7f361c67376e7fd8003ab0887cbb70c4e0d8d0f738e4b05732fd8cb5da24fa5f1112e20ba3603d58a80873d542c3a85337b597f63fc7a89837909196a9f0823625af4e2c18cc5274b56").to_vec(),
+			hex!("80ffff808a66c19052add13a202bcd73b546ae0cb70544f166c4a469672c666f0a5f9d8a80b84c2df313e2e749ff7e47eee888d9a023ba0a14a59852f4526b3e4b93b6dcbc807310fd50a0ae630c15e9eb07bda831d6d0cb6044d53a3dafb68e3fdb199fffdf80015ecd5e8af66e3d72ee5cc828c25989ca848e55396cccd9c196a4df1349fb9980a2ec48e449c43cc34954836cc14af398695f6569e301cef5a13eb88a16aa395580fd068d1339506db2893ba54a00a85aa712d68ff98ceeb5f4632f4e53618bb77880a3f173abac33e571e2a66f13127eeec3fb31bb1ae6f4b0fca8e658bbfbb5e52a803084ef6eaf38b821c59b3de92c4679117509b0b031e52ef5a80fdcff72e498ec804f36b8fb07a75463165f1714181009c86a2790685e78abd43220f5ecb194c887802559300c82eef4b21724bba2706cc2815e98cac3993c8d8dc9057b1aaf45ae8d8006571e929d492077d682dbc911934874ec00335029a90bd39e37d6d641e11873800397abe2ea62a374e5f0650c54fb99e8ef825066da798b0f4d729a7281f3575880b35cfb12f77988e1305ba651db7a8efbd43e4e8b057a56736cf6485f3033f481809ab1e406503e3ce63425a294f3f37aa4827b6a4ab38cf7e960a0d3335d79234e80be7c96632aa67491005e607a53bc1ab725fa465e29797bc29973d4cf5f64239b8037fe2a92c86d4c38ff38570b071e994ab86214e43e095dfb6ed142170fdac430").to_vec(),
+			hex!("be75edf06348b4330d1e88564111cb3d5bf19e4ed2927982e234d989e812f3f3010880ffe3135a9ee019dbfe4143608c9f4a4291ab827d7d9d055028d556d9cea2fce180eb5d42e7c6f84ac20e0d5ab42008c631b3e87f36d55d0d5053c9fb4f944ef97c").to_vec(),
+		].into_iter().into());
+		// Only interested in the DIP commitment verification part, we skip everything
+		// else.
+		let proof =
+			DipDidProofWithVerifiedStateRoot::<_, (), (), (), (), (), ()>::with_state_root_and_dip_commitment_proof(
+				parachain_state_root,
+				dip_commitment_proof,
+			);
+		assert_err!(
+			proof.verify_dip_commitment_proof_for_subject::<Keccak256, TestProviderRuntime>(
+				// We try
+				&AccountId32::from_ss58check("4qVtUbkD2xqp9cqGDjViPpFPesJNdfoJvGeSUgturBxAPyBK").unwrap(),
+			),
+			Error::DipCommitmentMerkleProof(MerkleProofError::InvalidProof)
+		);
+	}
+
+	#[test]
+	fn verify_dip_commitment_proof_for_subject_different_subject() {
+		// Storage proof generated at Peregrine block `5_258_991` with hash
+		// `0xd83da28e40d6e193af832916d2741252955d438c9d17a9f441279085db3e8daf` for
+		// storage key
+		// `0xb375edf06348b4330d1e88564111cb3d5bf19e4ed2927982e234d989e812f3f30d1c9ddb0ba4c0507243fb24936031b342ed12d34ad897938af6739bf519bdc2f101d67bf7dac85501a12dfa1fa4ab9a0000`
```

Comment:
It is just a question to understand the comment. On [line 306](https://github.com/KILTprotocol/kilt-node/pull/611/files/137bbdee9a9836b07d6610e4ce6644f7ba0c493d#diff-2d510a03a306a4788b824f796327df2510236b593467ece7fbf35ab6dcd5c65aR306), you generate the proof_verification_result for this account: &AccountId32::from_ss58check("4qVtUbkD2xqp9cqGDjViPpFPesJNdfoJvGeSUgturBxAPyBK").

Does this account have this storage key: 0xb375edf06348b4330d1e88564111cb3d5bf19e4ed2927982e234d989e812f3f30d1c9ddb0ba4c0507243fb24936031b342ed12d34ad897938af6739bf519bdc2f101d67bf7dac85501a12dfa1fa4ab9a0000?

Reply:
The subject for all the proofs is always `4qVtUbkD2xqp9cqGDjViPpFPesJNdfoJvGeSUgturBxAPyBK`, and only once is `4pebirGcQAJ4nyd5137VuK8TPVW9RXprWvZLQK1wcw2qJvnM` when we test for the wrong subject case.

So subject `4qVtUbkD2xqp9cqGDjViPpFPesJNdfoJvGeSUgturBxAPyBK` has the storage key `0xb375edf06348b4330d1e88564111cb3d5bf19e4ed2927982e234d989e812f3f30d1c9ddb0ba4c0507243fb24936031b342ed12d34ad897938af6739bf519bdc2f101d67bf7dac85501a12dfa1fa4ab9a0000`, while subject `4pebirGcQAJ4nyd5137VuK8TPVW9RXprWvZLQK1wcw2qJvnM` has the storage key `0xb375edf06348b4330d1e88564111cb3d5bf19e4ed2927982e234d989e812f3f346802a0d131133fa4cac8e6332f14ad28fe8b2ccb9e339f1c36798e918846726e6e983b59dd4fa3101a12dfa1fa4ab9a0000`.

Reply:
So what you have commented with is not correct, since the storage `0xb375edf06348b4330d1e88564111cb3d5bf19e4ed2927982e234d989e812f3f30d1c9ddb0ba4c0507243fb24936031b342ed12d34ad897938af6739bf519bdc2f101d67bf7dac85501a12dfa1fa4ab9a0000` is for subject `4qVtUbkD2xqp9cqGDjViPpFPesJNdfoJvGeSUgturBxAPyBK`.

Reply:
Okay, thanks for the clarification